src/main/kotlin/com/terraformation/backend/customer/OrganizationService.kt

```diff
@@ -77,6 +78,47 @@ class OrganizationService(
     }
   }
 
+  /**
+   * Assigns a Terraformation Contact in an organization. Removes existing Terraformation Contact
+   * from the organization, if one exists. If email of user to assign as Terraformation Contact,
+   * already exists as an organization user, the role is simply updated. Otherwise, a new user is
+   * created and added as the Terraformation Contact.
+   *
+   * @param email, email of user to assign as Terraformation Contact
+   * @param organizationId, organization in which to assign the Terraformation Contact
+   * @return id of user that was assigned as Terraformation Contact
+   */
+  fun assignTerraformationContact(email: String, organizationId: OrganizationId): UserId {
+    return dslContext.transactionResult { _ ->
+      val currentTfContactUserId = organizationStore.fetchTerraformationContact(organizationId)
+      if (currentTfContactUserId != null) {
+        organizationStore.removeUser(organizationId, currentTfContactUserId)
+      }
+      val existingUser = userStore.fetchByEmail(email)
+      val orgUserExists =
+          if (existingUser != null) {
+            dslContext
+                .selectOne()
+                .from(ORGANIZATION_USERS)
+                .where(ORGANIZATION_USERS.ORGANIZATION_ID.eq(organizationId))
+                .and(ORGANIZATION_USERS.USER_ID.eq(existingUser.userId))
+                .fetch()
+                .isNotEmpty
+          } else {
+            false
```

Comment:
I think this should throw an exception. Can't see a good use case for wanting to create a new TF employee user as part of the "assign someone as contact" flow; more likely would be a typo in the email address.

Reply:
That's a valid point, will update this.